packages/blade/src/components/BottomSheet/_decisions/decisions.md

```diff
@@ -0,0 +1,225 @@
+
+1. We coupled the BottomSheet & SelectInput tightly and internally conditionally switch the components
+
+Pros:
+
+- Easy implementation for user's end
+- User's don't have to think about breakpoints, conditional rendering etc
+- From blade side, we will have greater control over the pattern
+
+Cons:
+
+- Bundle size will be impacted, even if users are desktop they will get the bundle of BottomSheet (vice versa)
+
+2. We expose BottomSheet indepandantly and let users lazy load the component as needed
+
+Pros:
+
+- No uneccesary bundle size impact for any of the platforms
+
+Cons:
+
+- Not trivial to implement from user's end, they will have to compose the BottomSheet & Select as per their needs.
+
+
+Considering the bundle size downside to approach 1, we decided to go ahead with approach 2.
+
+### Composition Example
+
+```jsx
+const BottomSheet = React.lazy();
+const DropdownOverlay = React.lazy();
+
+const SelectContent = () => {
+  return (
+    <>
+      <ActionListItem
+        leading={<ActionListItemIcon icon={SettingsIcon} />}
+        title="Settings"
+        value="settings"
+      />
+      <ActionListItem leading={<ActionListItemIcon icon={InfoIcon} />} title="Info" value="info" />
+    </>
+  );
+};
+
+const App = () => {
+  const [open, setOpen] = useState(false);
+
+  return (
+    <Dropdown selectionType={selectionType}>
+      <SelectInput label="Select Action" onClick={() => setOpen(true)} />
+      {isMobile ? (
+        <BottomSheet open={open} onDismiss={() => setOpen(false)}>
+          <BottomSheetHeader>
+            <BottomSheetHeaderLeading title="Payments Links" prefix={PayIcon} />
+          </BottomSheetHeader>
+          <BottomSheetBody>
+            <SelectContent />
+          </BottomSheetBody>
+          <BottomSheet>
+            <BottomSheetLeading title="Footer Title" prefix={icon} />
+            <BottomSheetTrailing
+              actions={{
+                primaryAction: { text: 'Confirm' },
+                secondaryAction: { text: 'Close' },
+              }}
+            />
+          </BottomSheet>
+        </BottomSheet>
+      ) : (
+        <DropdownOverlay>
+          <SelectContent />
+        </DropdownOverlay>
+      )}
+    </Dropdown>
+  );
+};
+```
+
+## Accessibility
+
+The bottom sheet will follow the accessibility charechtaristics of a Modal.
+
+[APG Guidelines for Modal](https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/)
+
+Behaviours:
+
+- Tab, Shift+Tab should cycle the focused elements
+- The focus should be trapped inside the bottom sheet
+- Escape closes the bottom sheet, after closing the focus will go to the triggered element
+- When opening the bottom sheet it should either focus on the first focusable element on the sheet or element provided by the user via `initialFocusRef` prop
+
+## Open questions
+
+1. What is `action` in the header trailing visual?
+2. In BottomSheetHeaderLeaing & BottomSheetFooterLeading will the `prefix` only support Icon component?
```

Comment:
Are these resolved? Let's add the resolutions.

Reply:
@saurav12 can you check these? 

1. What is `action` in the header trailing visual?

<img width="657" alt="image" src="https://user-images.githubusercontent.com/35374649/221138260-83dfc66c-6117-4fbf-965e-e712c020c9a2.png">

2. In BottomSheetHeaderLeading & BottomSheetFooterLeading will the `prefix` only support Icon component?

Reply:
bumping this